### ISSUE TITLE
Newsletter\Model\Subscriber::loadByEmail() does not use MySQL index

### ISSUE THREAD

**Issue by schmengler** [magento/magento2#12787]
### Preconditions
1. Any Magento version between 1.0 and 2.2

### Steps to reproduce
1. Call `$subscriber->loadByEmail($email)` with debugger or profiler
2. Inspect SQL query
3. Inspect execution plan `EXPLAIN SELECT * FROM newsletter_subscriber WHERE subscriber_email='...'`

### Expected result

1. An index should be use to retrieve the subscriber instantly

### Actual result
1. No index is used, resulting in a slow "Using where" query like this:
```
+----+-------------+-----------------------+------+---------------+------+---------+------+-------+-------------+
| id | select_type | table                 | type | possible_keys | key  | key_len | ref  | rows  | Extra       |
+----+-------------+-----------------------+------+---------------+------+---------+------+-------+-------------+
|  1 | SIMPLE      | newsletter_subscriber | ALL  | NULL          | NULL | NULL    | NULL | 75735 | Using where |
+----+-------------+-----------------------+------+---------------+------+---------+------+-------+-------------+
```

**Comment by devamitbera** [comment]
#mm18in

**Comment by magento-engcom-team** [comment]
@devamitbera thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by ishakhsuvarov** [comment]
Hi @schmengler. Thank you for your report. 
The issue has been fixed in magento/magento2#13033 by @devamitbera in 2.2-develop branch
Related commit(s):
 - 755978397a8d0bacbfe81f7240b1e3ff3b9b23af
 - b85349745c4a593306edf61dae2235cd96a0fcd2
 - be8b258b6a7ff406f099619c917e2001731f1157

The fix will be available with the upcoming patch release.

**Comment by magento-engcom-team** [comment]
Hi @schmengler. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1351 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 70dd5a3fa17e1ee1cdfaf7f0defd9276989ebbd0
 - 54bc3a1dca861e11e46e12f5a15578e0cb7ceb6d
 - 3debd3f87e27e23949b81cb2d7e881b5a269608c
 - 72c496566bf0d2ff1ced0401514faa5cbaccc0f8
 - e60ecdfeabe8b8839cf52fd5ba7fefcbb0b9bdc3
 - 8703b3da3dbe28f71c1daeaba08bbe44887f0733
 - bd88ed8cc4e1f98b791bd7ca8bed45c29b276ee1
 - 1c70b18a339fddf8a51a476a5a251aabef951350
 - 5fd7481d2cf8b298369837508a16a23ab4f49850
 - 7c615065b36cdafa95d752a891c09dd4ab1a4dfa
 - 2653b8ae55625637a6836d305451e292f881bd6a
 - 917a03f89f98454cb2ecde524dc8557d7fdcc135
 - eca73d0abd1ca6e110c364ef380adab86ab3cc1a
 - 26f3109842ce89096a49b9c122af0bae6b672ba2
 - c223484a75e38fdcece09d4586342528f06c6383
 - 3b90ad71d791ba9cf8e81cc204fd306f41473a18
 - ad317c25039de70c1a976e1139c57730899eb114
 - 53e721ccc06f1ade05132690715575bd7d47d6b7
 - 6114120c0a921b342871448dfb6c47bf34a65cdc
 - 63b483925c18304a32441f66b0db8a9c70995c73

The fix will be available with the upcoming 2.3.0 release.
